test/integration/performance/connection_http_transport_benchmark_test.py

```diff
@@ -0,0 +1,264 @@
+    def execute_import_command(self):
+        pass
+
+    def perform_import(self):
+        self.execute_import_command()
+        count_query = f"SELECT count(*) FROM {self.import_into_table_name};"
+        return self.connection.execute(count_query).fetchval()
+
+
+class ImportCsvBenchmark(ImportBenchmark):
+    @property
+    def source_file(self):
+        return "sample_import.csv"
+
+    def prepare_data(self):
+        # save some time by only exporting the file once
+        if self.source_filepath.is_file():
+            return
+        self.connection.export_to_file(
+            self.source_filepath,
+            f"SELECT SALES_TIMESTAMP, PRICE, CUSTOMER_NAME FROM {self.export_from_table_name}",
+        )
+
+    def execute_import_command(self):
+        self.connection.import_from_file(
+            self.source_filepath,
+            table=self.import_into_table_name,
+            import_params={"columns": ["SALES_TIMESTAMP", "PRICE", "CUSTOMER_NAME"]},
+        )
+
+
+# schema = pa.schema([
+#     ("SALES_TIMESTAMP", pa.timestamp("ms")),
+#     ("PRICE", pa.float64()),
+#     ("CUSTOMER_NAME", pa.string())
+# ])
+#
+
+
+class ImportParquetBenchmark(ImportBenchmark):
+    @property
+    def source_file(self):
+        return "sample_import.parquet"
+
+    def prepare_data(self):
+        # save some time by only exporting the file once
+        # bit more complicated as we don't already have an export to parquet,
+        # not sure how "optimized" it should be here. could do to file & stream to build it?
+        if self.source_filepath.is_file():
+            return
+
+        # with pq.ParquetWriter(self.source_file, schema) as writer:
+        #     for batch in data_generator():
+        #         table = pa.Table.from_pydict(batch, schema=schema)
+        #         writer.write_table(table)
+        # self.connection.export_to_file(
+        #     self.source_filepath,
+        #     f"-- SELECT SALES_TIMESTAMP, PRICE, CUSTOMER_NAME FROM {self.export_from_table_name}",
+        # )
+
+    def execute_import_command(self):
+        self.connection.import_from_parquet(
+            source=self.source_filepath,
+            table=self.import_into_table_name,
+        )
+
+
+@pytest.mark.parametrize(
+    "import_class",
+    (
+        pytest.param(ImportCsvBenchmark, id="csv"),
+        # pytest.param(ImportParquetBenchmark, id="parquet"),
+    ),
+)
+@pytest.mark.benchmark
+def test_import(
+    benchmark,
+    import_class: ImportBenchmark,
+    connection: ExaConnection,
+    tmp_source_directory: Path,
+    create_initial_sales_data,
+):
+    # currently, connection runs with NO_CERT & WITH_CERT, for local testing only
+    # NO_CERT was used as WITH_CERT adds the complexity of a VM. considering
+    # our main concern is the performance of the import functions, we might want to
+    # only consider NO_CERT.
+
+    def setup():
+        # Reset state here; need at 0 so data doesn't increase throughout runs
```

Comment:
not sure how important this is to reset table each time.
when this was initially added, i was using "SALES" for everything.
now that we have "SALES", and "TMP_SALES_COPY", we could, assuming no DB changes, continue to add on to TMP_SALES_COPY -> only thing is it is useful to verify that the code works as desired by checking the output rows (which is we keep adding on to TMP_SALES_COPY makes is a function of iterations.

Reply:
chose to leave as is as not affecting and otherwise we'd have to use pytest-benchmark setup. as within our single iteration, the rounds do not  reset function-scoped fixtures.